src/Metadata/FluentConfiguration/EntityTypeBuilder.cs

```diff
@@ -60,40 +60,62 @@ public EntityTypeBuilder<TEntity> HasTableName(string tableName)
         /// <typeparam name="TProperty">Propery type.</typeparam>
         /// <param name="propertyExpression">A lambda expression representing the property to be configured.</param>
         /// <returns>An object that can be used to configure the property.</returns>
-        public virtual PropertyBuilder<TEntity> Property<TProperty>(
+        IPropertyBuilder<TEntity> IEntityPropertyBuilder<TEntity>.Property<TProperty>(
             Expression<Func<TEntity, TProperty>> propertyExpression)
         {
+            Check.NotNull(propertyExpression, nameof(propertyExpression));
             string propertyName = PropertyName<TEntity>.GetPropertyName(propertyExpression);
-
-            if (!_propertyBuilders.TryGetValue(propertyName, out PropertyBuilder<TEntity> propertyBuilder))
+            if (_propertyBuilders.ContainsKey(propertyName))
             {
-                propertyBuilder = new PropertyBuilder<TEntity>(this, propertyName);
-                _propertyBuilders.Add(propertyName, propertyBuilder);
+                throw new InvalidOperationException(string.Format("Property \"{0}\" was already configured.", propertyName));
```

Comment:
Is necessary throws exception?
What when user try something like:
```csharp
modelBuilder.Entity<Foo>().Property(p=> p.Property1).HasColumnName("Name");
modelBuilder.Entity<Foo>().Property(p=> p.Property1).UseConverter<AddressConverter>();
```

Reply:
Yes. I was thinking about it. But I want to force to use "one-line" setup for properties. That way it is not possible to make configuration, which does not make sense. For example:

``` csharp
modelBuilder.Entity<Foo>().Property(p=> p.Property1).UseConverter<AddressConverter>();
modelBuilder.Entity<Foo>().Property(p=> p.Property1).IgnoreConverter();
modelBuilder.Entity<Foo>().Property(p=> p.Property1).NoMap();
```